src/api.js

```diff
@@ -1650,9 +1650,20 @@ api.declare({
   }
 
   // Create temporary credentials for the task
+  let clientId = [
+    'task-client',
+    taskId,
+    `${runId}`,
+    'on',
+    run.workerGroup,
+    run.workerId,
+    'until',
+    `${takenUntil.getTime() / 1000}`,
+  ].join('/');
```

Comment:
Let's pull this out to a utility function since it now appears in two places?

Reply:
You're quite right. There might be other common code between `claimTask` and `reclaimTask` that could be shared too - I'll take a look.